packages/core/src/database/index.ts

```diff
@@ -654,202 +649,8 @@ export const createDatabase = async ({
-
-      if (hasPonderMetaTable) {
-        await this.wrap({ method: "migrate", includeTraceLogs: true }, () =>
-          qb.migrate.transaction().execute(async (tx) => {
-            const previousApps = await tx
-              .selectFrom("_ponder_meta")
-              // @ts-ignore
-              .where("key", "like", "app_%")
-              .select("value")
-              .execute()
-              .then((rows) => rows.map(({ value }) => value as PonderApp));
-
-            if (
-              previousApps.some(
-                (app) =>
-                  app.is_locked === 1 &&
-                  app.heartbeat_at + common.options.databaseHeartbeatTimeout >
-                    Date.now(),
-              )
-            ) {
-              throw new NonRetryableError(
-                `Migration failed: Schema '${namespace}' has an active app`,
-              );
-            }
-
-            for (const app of previousApps) {
-              for (const table of app.table_names) {
-                await tx.schema
-                  // @ts-ignore
-                  .dropTable(`${app.instance_id}__${table}`)
-                  .cascade()
-                  .ifExists()
-                  .execute();
-                await tx.schema
-                  // @ts-ignore
-                  .dropTable(`${app.instance_id}_reorg__${table}`)
-                  .cascade()
-                  .ifExists()
-                  .execute();
-              }
-              await tx
-                .deleteFrom("_ponder_meta")
-                // @ts-ignore
-                .where("key", "=", `status_${app.instance_id}`)
-                .execute();
-              await tx
-                .deleteFrom("_ponder_meta")
-                // @ts-ignore
-                .where("key", "=", `app_${app.instance_id}`)
-                .execute();
-            }
-
-            if (previousApps.length > 0) {
-              common.logger.debug({
-                service: "database",
-                msg: "Migrated previous app to v0.8",
-              });
-            }
-          }),
-        );
-      }
-
-      // 0.9 migration
-
-      if (hasPonderMetaTable) {
-        await qb.migrate
-          .deleteFrom("_ponder_meta")
-          // @ts-ignore
-          .where("key", "=", "status")
-          .execute();
-
-        const version: string | undefined = await qb.migrate
-          .selectFrom("_ponder_meta")
-          .select("value")
-          .where("key", "=", "app")
-          .executeTakeFirst()
-          .then((row) => row?.value.version);
-
-        if (version === undefined || Number(version) < Number(VERSION)) {
-          await qb.migrate.schema
-            .dropTable("_ponder_status")
-            .ifExists()
-            .cascade()
-            .execute();
-        }
-      }
-
       await this.wrap(
-        { method: "migrate", includeTraceLogs: true },
+        { method: "createPonderSystemTables", includeTraceLogs: true },
```

Comment:
Should these statements use a transaction?

Reply:
The `IF NOT EXISTS` or `OR REPLACE` make all the statements in this block idempotent